csrc/ir/internal_base_nodes.h

```diff
@@ -529,22 +529,41 @@ class TensorDomain : public Val {
     return root_domain_.empty() ? logical_domain_ : root_domain_;
   };
 
+  bool isRoot(const IterDomain* id) const {
```

Comment:
We might want to include a comment here indicating that this is always false if there is no root domain, i.e. this is not the same as checking `getMaybeRootDomain()` for `id`. Same goes for loop and allocation.